internal/services/shared/errors.go

```diff
@@ -129,6 +129,13 @@ func RewriteError(ctx context.Context, err error) error {
 	case errors.Is(err, context.DeadlineExceeded):
 		return status.Errorf(codes.DeadlineExceeded, "%s", err)
 	case errors.Is(err, context.Canceled):
+		err := context.Cause(ctx)
```

Comment:
TIL there is actually a cause to cancellation

Reply:
As of Go 1.20